### PR TITLE
Coding Intro

### PR DESCRIPTION
**Pull request by kapnobatai137**
TODOs:
- [x] finish loops workout
- [x] write practice questions
- [x] write revision questions
- [x] split long insights
- [x] replace syntax footnotes (in other insights) with the glossary
- [x] better icon

### CODE REVIEW

**Comment by enki-hq** [comment]
Failed parsing files: 
  - programming-fundamentals/core/functions/conditions-ii.md: 
 > 	No valid answer list in question node
  - programming-fundamentals/core/introduction/programming-benefits.md: 
 > 	No valid answer list in question node

**Comment by enki-hq** [comment]
Failed parsing files: 
  - programming-fundamentals/core/introduction/intro-to-programming.md: 
 > 	No valid answer list in question node
  - programming-fundamentals/core/introduction/programming-benefits.md: 
 > 	No valid answer list in question node
  - programming-fundamentals/core/introduction/why-learn-to-code.md: 
 > 	No valid answer list in question node

**Comment by enki-hq** [comment]
Failed parsing files: 
  - programming-fundamentals/core/introduction/intro-to-programming.md: 
 > 	No valid answer list in question node
  - programming-fundamentals/core/introduction/programming-benefits.md: 
 > 	No valid answer list in question node
  - programming-fundamentals/core/introduction/why-learn-to-code.md: 
 > 	No valid answer list in question node

File: programming-fundamentals/core/data/data-types.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+Data types are very important! They decide what actions you can take on that piece of data. One good example is that you can't add the number `1` to the text `"one"`.
+
+---
+## Practice
+
+The most common data types are:
+- numerical
+- text-based
+- lists
```
Comment (kapnobatai137) [inline programming-fundamentals/core/data/data-types.md:35]: this is not a mistake, I'm trying a new type of "question" which is actually a reminder of the most important concept in the previous insight

**Comment by enki-hq** [comment]
Failed parsing files: 
  - programming-fundamentals/core/introduction/intro-to-programming.md: 
 > 	No valid answer list in question node
  - programming-fundamentals/core/introduction/programming-benefits.md: 
 > 	No valid answer list in question node
  - programming-fundamentals/core/introduction/why-learn-to-code.md: 
 > 	No valid answer list in question node

**Comment by enki-hq** [comment]
Failed parsing files: 
  - programming-fundamentals/core/introduction/intro-to-programming.md: 
 > 	No valid answer list in question node
  - programming-fundamentals/core/introduction/programming-benefits.md: 
 > 	No valid answer list in question node
  - programming-fundamentals/core/introduction/why-learn-to-code.md: 
 > 	No valid answer list in question node

**Comment by enki-hq** [comment]
Rule insight:slug failed for files: 
  - programming-fundamentals/core/data/lists.md
  - programming-fundamentals/core/data/data-types.md

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/programming-fundamentals/core/data/integers.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/data/integers.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/data/pf-data-types.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/data/pf-data-types.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/data/pf-lists.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/data/pf-lists.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/data/strings.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/data/strings.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/data/variables-i.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/data/variables-i.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/data/variables-ii.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/data/variables-ii.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/functions/conditions-i.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/functions/conditions-i.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/functions/conditions-ii.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/functions/conditions-ii.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/functions/functions-i.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/functions/functions-i.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/functions/functions-ii.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/functions/functions-ii.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/functions/functions-iii.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/functions/functions-iii.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/introduction/hello-code.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/introduction/hello-code.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/introduction/intro-to-programming-takeaway.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/introduction/intro-to-programming-takeaway.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/introduction/intro-to-programming.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/introduction/intro-to-programming.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/introduction/programming-benefits-takeaway.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/introduction/programming-benefits-takeaway.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/introduction/programming-benefits.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/introduction/programming-benefits.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/introduction/why-learn-to-code-takeaway.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/introduction/why-learn-to-code-takeaway.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/introduction/why-learn-to-code.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/introduction/why-learn-to-code.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/loops/condition-loops.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/loops/condition-loops.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/loops/counter-loops.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/loops/counter-loops.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)
- [ ] [enkidevs/curriculum/programming-fundamentals/core/loops/loops.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/programming-fundamentals/core/loops/loops.md?ref=6227b717b23a999e07d2d04a2d80f6917c887145)

**Comment by enki-hq** [comment]
Rule insight:slug failed for files: 
  - programming-fundamentals/core/functions/debugging.md

**Comment by enki-hq** [comment]
Rule insight:aspects failed for files: 
  - programming-fundamentals/core/loops/counter-loops.md

**Comment by enki-hq** [comment]
Rule insight:footnotes failed for files: 
  - coding-intro/core/functions/first-function.md

**Pull request merged**